custom_components/iaqualink_robots/vacuum.py

```diff
@@ -258,137 +281,170 @@ async def async_update(self):
             self._device_type = data[index]["device_type"]
             self._attributes['device_type'] = self._device_type
 
+            #Set modes based on model
+            if self._device_type == "vr":
 
-        #request device status over websocket
-        request = { "action": "subscribe", "namespace": "authorization", "payload": { "userId": self.id }, "service": "Authorization", "target": self._serial_number, "version": 1 }
+                self._fan_speed_list = ["Floor only", "SMART Floor and walls", "Floor and walls"]
+                self._supported_features = SUPPORT_IAQUALINK_ROBOTS_vr
 
-        data= None
-        data = await asyncio.wait_for(self.get_device_status(request), timeout=30)
+            if self._device_type == "cyclonext":
 
-        try:
-            self._status = data['payload']['robot']['state']['reported']['aws']['status']
-            self._attributes['status'] = self._status
-        except:
-            #returns empty message somethimes, try second call
-            self._debug = data
-            if self._debug_mode == True:
-                self._attributes['debug'] = self._debug
-            
-            data = await asyncio.wait_for(self.get_device_status(request), timeout=30)
+                self._fan_speed_list = ["Floor only", "SMART Floor and walls"]
```

Comment:
This line should be `self._fan_speed_list = ["Floor only", "Floor and walls"]`, otherwise it breaks the option to change to mode 3 in cyclonext.